### ISSUE TITLE
Undownloadable search path entry should be an error

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nix#2081]
Currently I get a message like

```
warning: Nix search path entry 's3://blahblah/barbar' cannot be downloaded, ignoring
```

And then Nix proceeds to throw a different and misleading error because it used the wrong search path. @edolstra is there a particular reason you made that non-fatal? I would think that if someone goes out of their way to put a particular URL into their search path, they want to know if that URL is broken. Nix is also not very clear about the error that broke that download, which makes it harder to diagnose.

**Comment by copumpkin** [comment]
@edolstra I can switch it from warning to error if you're okay with that 😄but before changing it I'd rather understand if there's a use case I'd be breaking by doing that

**Comment by copumpkin** [comment]
@edolstra it seems like if someone explicitly passes in `-I`, they expect that to be respected. I could understand the warning behavior on `NIX_PATH` though. Would you object to me making `-I` stricter than the env var?

**Comment by edolstra** [comment]
No, but I'm not sure that users *expect* non-existent `-I` flags to be a fatal error. For example, GCC, Perl, Java, ... don't complain if you pass a non-existent directory on the command line.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
